service-node/backend/src/main/scala/org/enterprisedlt/fabric/service/node/rest/UploadHandler.scala

```diff
@@ -0,0 +1,90 @@
+import org.eclipse.jetty.server.Request
+import org.eclipse.jetty.server.handler.AbstractHandler
+import org.eclipse.jetty.util.IO
+import org.enterprisedlt.fabric.service.node.Util
+import org.slf4j.LoggerFactory
+
+import scala.util.Try
+
+/**
+ * @author Alexey Polubelov
+ */
+
+
+class UploadHandler(
+    multipartConfig: MultipartConfigElement,
+    endpoints: Array[FileUploadEndpoint],
+    createCodec: () => ServerCodec,
+) extends AbstractHandler {
+
+    private val logger = LoggerFactory.getLogger(this.getClass)
+
+    override def handle(target: String, baseRequest: Request, request: HttpServletRequest, response: HttpServletResponse): Unit = {
+        if (request.getMethod == "POST" && endpoints.exists(_.uri == request.getPathInfo)) {
+            RestEndpointContext.set(RestEndpointContext(request, response))
+            request.setAttribute(Request.MULTIPART_CONFIG_ELEMENT, multipartConfig)
+            processRequest(endpoints, request, response)
+            RestEndpointContext.clean()
+            baseRequest.setHandled(true)
+        }
+
+    }
+
+    private def processRequest(
+        endpoints: Array[FileUploadEndpoint],
+        request: HttpServletRequest, response: HttpServletResponse
+    ): Unit = {
+        if (endpoints.exists(_.uri == request.getPathInfo)) {
+            val storeLocation = endpoints
+              .filter(_.uri == request.getPathInfo)
+              .head
+              .location
+            Util.ensureDirExists(storeLocation)
+            processParts(request, response, Paths.get(storeLocation)) match {
+                case Right(_) => response.setStatus(HttpServletResponse.SC_OK)
+                case Left(msg) =>
+                    logger.info(s"Internal error: $msg")
+                    response.getWriter.println(s"Internal error: $msg")
+                    response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR)
+            }
+
+        }
+        else
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND)
+    }
+
+
+    private def processParts(request: HttpServletRequest, response: HttpServletResponse, outputDir: Path): Either[String, Unit] = {
+        Try {
+            request.getParts.forEach { part =>
+                val filename = part.getSubmittedFileName
+                if (Option(filename).nonEmpty) {
+                    logger.info(s"Got Part ${part.getName} with size = ${part.getSize}, contentType = ${part.getContentType}, submittedFileName $filename")
+                    val outputFile = outputDir.resolve(filename)
+                    val is = part.getInputStream
+                    val os = Files.newOutputStream(outputFile, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+                    IO.copy(is, os)
+                    logger.info(s"Saved Part ${part.getName} to ${outputFile.toString}")
+                    is.close()
+                    os.close()
+                }
+            }
+        }.toEither match {
```

Comment:
consider merge this match with one in calling method

Reply:
done.